orchagent/neighorch.cpp

```diff
@@ -137,16 +137,24 @@ void NeighOrch::doTask(Consumer &consumer)
         }
 
         string alias = key.substr(0, found);
-        Port p;
 
-        if (!gPortsOrch->getPort(alias, p))
+        if (alias == "eth0" || alias == "lo" || alias == "docker0")
         {
             it = consumer.m_toSync.erase(it);
             continue;
         }
 
+        Port p;
+        if (!gPortsOrch->getPort(alias, p))
+        {
+            SWSS_LOG_INFO("Port %s doesn't exist", alias.c_str());
+            it++;
+            continue;
```

Comment:
Not only the missing of the port will drop the neighbor message, but the missing of router interface will drop the neighbor message. according to the swss record, one example could be:
```
2017-05-17.19:18:11.538580|INTF_TABLE:PortChannel8:10.60.28.9/31|SET|scope:global|family:IPv4
2017-05-17.19:18:12.466388|LAG_TABLE:PortChannel8|SET|admin_status:up|oper_status:down|mtu:1500
2017-05-17.19:18:12.473995|LAG_MEMBER_TABLE:PortChannel8:Ethernet120|SET|status:disabled
2017-05-17.19:18:12.474591|LAG_MEMBER_TABLE:PortChannel8:Ethernet124|SET|status:disabled
2017-05-17.19:18:12.475460|LAG_TABLE:PortChannel8|SET|admin_status:up|oper_status:down|mtu:1500
2017-05-17.19:18:15.361938|LAG_MEMBER_TABLE:PortChannel8:Ethernet124|SET|status:enabled
2017-05-17.19:18:15.364166|LAG_MEMBER_TABLE:PortChannel8:Ethernet120|SET|status:enabled
2017-05-17.19:18:15.365244|LAG_TABLE:PortChannel8|SET|admin_status:up|oper_status:up|mtu:1500
2017-05-17.19:18:15.941662|NEIGH_TABLE:PortChannel8:10.60.28.8|SET|neigh:00:1c:73:3c:de:11|family:IPv4
```
In this case, the gap is very small and it is highly possible that the neighbor message is processed before the others and then get dropped.